### PR TITLE
RAC-5026 Add new MasterOVA jenkinfile with pipeline changes

### PR DESCRIPTION
**Pull request by patelb10**
RAC-5026
This PR is addressing fillowing things
We need  ability to run parts of the MasterCI test against development branches before they are merged to master.We should be able to do this in the test Jenkins sandbox.  
We started with creating a Post OVA test in the sandbox. It has two option run using pre-built OVA or Build and install OVA, then run FIT tests
This master OVA Pipeline runs only on test jenkins, updates in this PR allows to run  master pipeline without any changes.

### CODE REVIEW

**Comment by patelb10** [comment]
@hohene @johren @derrickostertag 
Successful Test Run at http://10.240.19.232:8080/blue/organizations/jenkins/Jeanne-MasterCI/detail/Jeanne-MasterCI/230/pipeline

**Review by hohene — APPROVED** [review]
Paired with Beena on this, +1

**Comment by derrickostertag** [comment]
+1

**Review by derrickostertag — APPROVED** [review]

File: jobs/FunctionTest/FunctionTest.groovy [hunk 1]
```diff
@@ -326,7 +330,7 @@ def dockerPostTest(TESTS, docker_stash_name, docker_stash_path, docker_record_st
 
 def ovaPostTest(TESTS, ova_stash_name, ova_stash_path, repo_dir, test_type){
     setOVA(ova_stash_name, ova_stash_path)
-    test_stack = "-stack vagrant"
+    test_stack = "-stack ova"
```
Comment (hohene) [inline jobs/FunctionTest/FunctionTest.groovy:333]: This isn't being correctly passed into the run_tests.py command line.
We should see python run_tests.py -test deploy/rackhd_stack_init.py -stack ova -xunit
```
17:04:07 17:04:07 [ova FIT] + '[' ucs == ucs ']'
17:04:07 17:04:07 [ova FIT] + '[' '-stack ova' == '-stack ****' ']'
17:04:07 17:04:07 [ova FIT] + python run_tests.py -test deploy/rackhd_stack_init.py -xunit
17:04:07 17:04:07 [ova FIT] deploy VM(s) using command argument
```
In test.sh, add quotes to the assignment command:
    tstack=${TEST_STACK}
should be:
   tstack="${TEST_STACK}"

Comment (hohene) [reply jobs/FunctionTest/FunctionTest.groovy:333]: +1

**Comment by hohene** [comment]
@PengTian0 @panpan0000

**Comment by panpan0000** [comment]
Thank you guys for those.
Can you please add more background in PR description ?
* Use Case
* What's the difference than OVA-Post-Test ? this seems run FIT against a pre-build OVA.
* what's the impact to current pipeline ?

**Review by stuart-stanley — APPROVED** [review]
Approved if @panpan0000 feels he had his questions answered. I will merge on his thumbs up.

**Comment by panpan0000** [comment]
I'm sorry, but where did those questions be answered ?

**Comment by anhou** [comment]
@stuart-stanley  it looks like @panpan0000 is questions is not clarified, then you've merged.  Is it an urgent PR?

**Comment by panpan0000** [comment]
Oh, I see. the PR description was updated.  My understanding is : The "Use Case" is "there's already a ova in place, I want to run all tests against it by pipeline."

talking with @PengTian0 , this PR functionality is okay. 
but I think the best way is to modify the https://github.com/RackHD/on-build-config/blob/master/jobs/build_ova/ova_post_test.groovy  ,  this groovy can be changed to accept a OVA as a input, and test it . which is more elegant in code architecture than create a standalone groovy.
so this kind of change can also apply to docker_post_test.groovy, vagrant_post_test.groovy as well..

**Comment by stuart-stanley** [comment]
I will look at those and open another story to shift them and/or talk with you both...

**Comment by hohene** [comment]
It is also to build an OVA from scratch using a 'modified' manifest file. We can point to a specific manifest flle to build from instead of from master.

**Pull request merged**